kpi/deployment_backends/kobocat_backend.py

```diff
@@ -674,6 +767,20 @@ def calculated_submission_count(self, requesting_user_id, **kwargs):
                                                       **kwargs)
         return MongoHelper.get_count(self.mongo_userform_id, **params)
 
+    def __get_latest_duplicate_submission(
```

Comment:
Would it be fair to say that this doesn't have anything to do with the submission being a duplicate anymore? It gets a single submission by UUID in JSON format, after validating something about the requesting user (I'm still tracing through `validate_submission_list_params()` to understand exactly what's going on there)

Reply:
It would be fair to say that 👍 probably closer to something like:

```python
get_submission_by_uid(self, requesting_user_id: int, _uuid: str, format_type: str = INSTANCE_FORMAT_TYPE_JSON)
``` 

to make it more reusable?

Reply:
OK, I get it now. I suggest removing this method and replacing the call to it:
```python
            return self.__get_latest_duplicate_submission(
                requesting_user_id, _uuid
            )
```
…with:
```python
            return next(
                self.get_submissions(requesting_user_id, query={'_uuid': _uuid})
            )
```

If you agree, here's a patch you can `git apply`:
```diff
diff --git a/kpi/deployment_backends/kobocat_backend.py b/kpi/deployment_backends/kobocat_backend.py
index 6fead4ba..37c19a1f 100644
--- a/kpi/deployment_backends/kobocat_backend.py
+++ b/kpi/deployment_backends/kobocat_backend.py
@@ -704,8 +704,8 @@ class KobocatDeploymentBackend(BaseDeploymentBackend):
         )
 
         if kc_response.status_code == status.HTTP_201_CREATED:
-            return self.__get_latest_duplicate_submission(
-                requesting_user_id, _uuid
+            return next(
+                self.get_submissions(requesting_user_id, query={'_uuid': _uuid})
             )
         else:
             raise KobocatDuplicateSubmissionException
@@ -769,20 +769,6 @@ class KobocatDeploymentBackend(BaseDeploymentBackend):
                                                       **kwargs)
         return MongoHelper.get_count(self.mongo_userform_id, **params)
 
-    def __get_latest_duplicate_submission(
-        self, requesting_user_id: int, _uuid: str
-    ) -> dict:
-        """
-        Retrieves the most recent duplicated submission for an asset in JSON
-        format
-        """
-        kwargs = {'query': {'_uuid': _uuid}}
-        params = self.validate_submission_list_params(
-            requesting_user_id, **kwargs
-        )
-
-        return next(self.__get_submissions_in_json(**params))
-
     def __get_submissions_in_json(self, **params):
         """
         Retrieves instances directly from Mongo.
```

Reply:
I'd still advocate for removing this method because the `validate_submission_list_params()` is duplicated from `get_submissions()`, and you don't have to drop to the lower-level `__get_submissions_in_json()` to pass the UUID query